src/components/drawer/PluginComponentDetailPanel.vue

```diff
@@ -0,0 +1,97 @@
+<template>
+  <q-drawer
+      bordered
+      data-cy="component-detail-panel"
+      v-model="isVisible"
+      side="right">
+
+    <div class="row" v-if="selectedComponent">
+      <div class="col">
+        <q-card flat>
+          <q-card-actions align="right">
+            <q-btn
+                flat
+                data-cy="close-component-detail-panel"
+                @click="closeDrawer"
+                icon="fa-solid fa-xmark"></q-btn>
+          </q-card-actions>
+          <q-card-section>
+            <q-form>
+              <q-input
+                  :label="$t('name')"
+                  v-model="selectedComponent.name">
+              </q-input>
+              <template v-for="definedAttribute in selectedComponent.definition.definedAttributes"
+                        :key="definedAttribute.name">
+                <component-attribute-input
+                    :original-object="selectedComponent"
+                    :attribute="definedAttribute"
+                    @update:attribute="upsertAttribute"
+                >
+                </component-attribute-input>
+              </template>
+            </q-form>
+          </q-card-section>
+          <q-card-actions align="evenly">
+            <q-btn flat
+                   icon="fa-solid fa-floppy-disk"
+                   @click="save"
+                   :label="$t('save')">
+            </q-btn>
+            <q-btn
+                flat
+                @click="reset"
+                :label="$t('reset')"
+                icon="fa-solid fa-arrow-rotate-left">
+            </q-btn>
+          </q-card-actions>
+        </q-card>
+      </div>
+    </div>
+  </q-drawer>
+
+</template>
+
+<script setup>
+import {
+  ref, computed, onUnmounted, unref,
+} from 'vue';
+import ComponentAttributeInput from 'components/inputs/PluginComponentAttributeInputWrapper';
+import PluginComponentSelectionEvent from 'src/composables/events/PluginComponentSelectionEvent';
+
+const selectedComponent = ref();
+let origValue = null;
+const isVisible = computed(() => !!selectedComponent.value);
+
+const selectionSubscription = PluginComponentSelectionEvent.subscribe((newVal) => {
+  origValue = JSON.parse(JSON.stringify(newVal));
+  selectedComponent.value = newVal;
+});
+
+function closeDrawer() {
+  PluginComponentSelectionEvent.next(null);
+}
+// dummy function for now, will probably recieve more logic
+function save() {
+  closeDrawer();
+}
+
+onUnmounted(() => {
+  selectionSubscription.unsubscribe();
+});
+
+function upsertAttribute(newAttr) {
```

Comment:
upsert isn't a word in English.

Reply:
`upsertAttribute` -> `updateAttribute`

Reply:
Added doc to function. Upsert is the correct naming here.